### PR TITLE
improve password handling by adding validation and preventing navigation

### PR DESCRIPTION
**Pull request by BerniWittmann**
I tried to sign up and came upon the following issue when using a password that is too short (e.g. `1234`).
The app navigates to the game page yet displays the error that my password was to short. 

I improved this by introducing the following fixes:
1. Added validation for password length input fields to give direct feedback to the user
2. Correctly throw and catch an error when the signup fails due to the validation server side to prevent the premature navigation

### CODE REVIEW

**Comment by DavidGolodetsky** [comment]
Great catch. Yes frontend rules should be equal to backend rules. Interesting, though I cannot reproduce redirecting to the game page.
image

File: client/src/utils/validations.js [hunk 1]
```diff
@@ -6,6 +6,9 @@ export const tooLongField = (v) =>
 export const tooShortField = (v) =>
   v === '' || !!(v && v.length >= 2) || 'Field is too short';
 
+export const passwordField = (v) =>
```
Comment (DavidGolodetsky) [inline client/src/utils/validations.js:9]: There is already a validation rule `shortPassword`

Comment (DavidGolodetsky) [reply client/src/utils/validations.js:9]: just remove `shortPassword` cos it's not used anywhere

**Review by DavidGolodetsky — CHANGES_REQUESTED** [review]

**Comment by DavidGolodetsky** [comment]
Great catch! Thanx for your contribution!

File: client/src/utils/validations.js (hunk 1 above)
Comment (BerniWittmann) [reply client/src/utils/validations.js]: Ah must have overlooked it

Comment (DavidGolodetsky) [reply client/src/utils/validations.js]: np

**Pull request merged**